### PR TITLE
WT-6753 Assert various types of history store operations are expected

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -113,6 +113,18 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
...
+             *  1) Delete the record with a tombstone with WT_TS_NONE.
+             *  2) Update the record's stop time point if the prepared update written to the data
+             * store is committed.
+             */
+            WT_ASSERT(session,
+              !WT_IS_HS(S2BT(session)) ||
+                (upd_arg->type == WT_UPDATE_TOMBSTONE && upd_arg->start_ts == WT_TS_NONE &&
+                  upd_arg->next == NULL) ||
+                (upd_arg->type == WT_UPDATE_TOMBSTONE && upd_arg->next != NULL &&
+                  upd_arg->next->type == WT_UPDATE_STANDARD && upd_arg->next->next == NULL));
```
Comment (quchenhao) [inline src/btree/row_modify.c:127]: To verify we only do this if previously we don't have a stop time point requires cracking the cell. I don't think we need to go that far.

**Review by hbokhari — COMMENTED** [review]
Is there a way to add these asserts somewhere inside `history` module? It looks a bit odd that we have a special debugging for history store record inside btree code.

**Comment by quchenhao** [comment]
> Is there a way to add these asserts somewhere inside `history` module? It looks a bit odd that we have a special debugging for history store record inside btree code.

Every history store operations goes through __hs_modify code but we can distinguish if they are inserts or updates there.

**Comment by quchenhao** [comment]
@hbokhari @bvpvamsikrishna Do you mind have another look on this one?

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Review by hbokhari — APPROVED** [review]
LGTM

**Pull request merged**